src/Analyzers/CSharp/CodeFixes/UseCollectionInitializer/CSharpUseCollectionInitializerCodeFixProvider.cs

```diff
@@ -140,42 +148,48 @@ private static ExpressionSyntax ConvertInvocation(InvocationExpressionSyntax inv
             if (!sourceText.AreOnSameLine(objectCreation.GetFirstToken(), objectCreation.GetLastToken()))
                 return true;
 
-            foreach (var match in matches)
-            {
-                var expression = GetExpression(match);
-
-                // If we have anything like: `new Dictionary<X,Y> { { A, B }, { C, D } }` then always make multiline.
-                // Similarly, if we have `new Dictionary<X,Y> { [A] = B }`.
-                if (expression is InitializerExpressionSyntax or AssignmentExpressionSyntax)
-                    return true;
-
-                // if any of the expressions we're adding are multiline, then make things multiline.
-                if (!sourceText.AreOnSameLine(expression.GetFirstToken(), expression.GetLastToken()))
-                    return true;
-            }
-
             var totalLength = "{}".Length;
             foreach (var match in matches)
```

Comment:
note: while this is getting cleaned up here, it is also getting refactored heavily in the next PR where i redo how we do formatting here entirely.  so don't have to examien this *too* closely :)